etl/src/energuide/embedded/ceiling.py

```diff
@@ -28,13 +28,13 @@ def from_data(cls, ceiling: element.Element) -> 'Ceiling':
                     french=ceiling.get_text('Construction/Type/French'),
                 ),
                 nominal_insulation=insulation.Insulation(
-                    float(ceiling.xpath('Construction/CeilingType/@nominalInsulation')[0])),
+                    ceiling.get('Construction/CeilingType/@nominalInsulation', float)),
                 effective_insulation=insulation.Insulation(
-                    float(ceiling.xpath('Construction/CeilingType/@rValue')[0])),
-                ceiling_area=area.Area(float(ceiling.xpath('Measurements/@area')[0])),
-                ceiling_length=distance.Distance(float(ceiling.xpath('Measurements/@length')[0])),
+                    ceiling.get('Construction/CeilingType/@rValue', float)),
+                ceiling_area=area.Area(ceiling.get('Measurements/@area', float)),
+                ceiling_length=distance.Distance(ceiling.get('Measurements/@length', float)),
             )
-        except (IndexError, ValueError, AssertionError) as exc:
+        except (ElementGetValueError, AssertionError) as exc:
```

Comment:
If you replace `element.get_text` with `element.get('...', str)` you can also do away with the `AssertionError`. Alternatively (perhaps better) we could update the `assert` with an explicit raise of `ElementGetValueError`.